### PR TITLE
closes #587

### LINKED ISSUE #587: Add status field to model spreadsheets to help manage property/class/constraint definitions at different levels of maturity
Right now the tools parse all tabs in a spreadsheet and generates RDF classes/properties/shapes/individuals for all records conforming to the expected column names.

It would be desirable to be able to flag specific properties, classes, shapes, individuals, and shape constraints to be ignored by the tools in some scenarios - say for example if the property/class/individual/constraint is incompletely defined or has not be approved by governance.

It is also desirable to be able to indicate to the build what level of status to use.  One build may utilize only terms/constraints that have a "stable" status, whereas another build may also include terms that have an "experimental" or "proposed" status.

This issue is focused only on adding the ability to label Property, Class and SHACL Property constraints with URI values.  We'll add other issues to address the other desiderata mentioned above.

### Acceptance Criteria

* Add a class named `io.konig.core.vocab.XOWL` which contains the following static field:

```
   public static final URI termStatus = new URIImpl("https://schema.pearson.com/ns/xowl/termStatus");
```

* Update the class `io.konig.shacl,PropertyConstraint` by adding:
   *  A field named `termStatus` whose value is a URI.
   * Setter and getter methods for `termStatus`
* Update the class `io.konig.spreadsheet.WorkbookLoader` as follows:
   *   Add a field `private static final String TERM_STATUS = "Term Status";`
* Update the class `io.konig.spreadsheet.WorkbookLoader.Worker` as follows:
   * Add the field `private int pcTermStatusCol` which stores the index for the column labeled `Term Status`.
   * Set the value of the `pcTermStatusCol` field in the `readPropertyConstraintHeader` method
   * In the method `loadPropertyConstraintRow` parse the contents of the `Term Status` column using the `uriValue(Row, int)` method.  If the value is non-null, annotate the `constraint` by invoking `edge(constraint, XOWL.termStatus, termStatusValue)`
* Use a similar design pattern to annotate OWL Class and RDF Property resources with `termStatus` properties.
* Add a test case to `io,konig.spreadsheet.WorkbookLoaderTest` to verify that the above changes work as expected.

### PR DESCRIPTION
**Pull request by Uvashriamm**

### CODE REVIEW

File: konig-core/src/main/java/io/konig/core/vocab/XOWL.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * limitations under the License.
+ * #L%
+ */
+
+
+import org.openrdf.model.URI;
+import org.openrdf.model.impl.URIImpl;
+
+public class XOWL {
+	public static final URI TERMSTATUS = new URIImpl("http://schema.pearson.com/ns/xowl/termstatus");
```
Comment (gmcfall) [inline konig-core/src/main/java/io/konig/core/vocab/XOWL.java]: We are using the camelCase convention for property names.

Please replace `termstatus` with `termStatus` so that the fully-qualified URI is
```
http://schema.pearson.com/ns/xowl/termStatus
```
  
Also, please replace the static field name `TERMSTATUS` with `termStatus`.

File: konig-spreadsheet/src/main/java/io/konig/spreadsheet/WorkbookLoader.java [hunk 2]
```diff
@@ -1243,7 +1249,9 @@ private void loadPropertyConstraintRow(Row row) throws SpreadsheetException {
 			if (partitionOf != null) {
 				pathHandlers.add(new PathHandler(constraint, Konig.partitionOf, partitionOf));
 			}
-
+			if(termStatus != null) {
```
Comment (gmcfall) [inline konig-spreadsheet/src/main/java/io/konig/spreadsheet/WorkbookLoader.java]: You don't need to test whether termStatus is null.  The null check happens inside the `edge` method.

File: konig-spreadsheet/src/main/java/io/konig/spreadsheet/WorkbookLoader.java [hunk 3]
```diff
@@ -2032,9 +2048,16 @@ private void loadClassRow(Row row) throws SpreadsheetException {
 					for (URI subclassId : subclassOf) {
 						graph.edge(classId, RDFS.SUBCLASSOF, subclassId);
 						graph.edge(subclassId, RDF.TYPE, OWL.CLASS);
+						if(termStatus != null) {
+							graph.edge(classId, XOWL.TERMSTATUS, termStatus);
```
Comment (gmcfall) [inline konig-spreadsheet/src/main/java/io/konig/spreadsheet/WorkbookLoader.java]: This is superfluous.  Why add this edge inside the loop of subclasses?

It is sufficient to add the edge once outside the loop, as you do below.

File: konig-core/src/main/java/io/konig/core/vocab/XOWL.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	public static final URI Stable = new URIImpl("http://schema.pearson.com/ns/xowl/termstatus/Stable");
+	public static final URI Experimental = new URIImpl("http://schema.pearson.com/ns/xowl/termstatus/Experimental");
```
Comment (gmcfall) [inline konig-core/src/main/java/io/konig/core/vocab/XOWL.java]: Please replace the terms `Stable` and `Experimental` with:

* statusStable
* statusExperimental

And add:
* statusSubmitted
* statusInvalid
* statusSuperseded
* statusRetired

These terms are defined in the document [Term Status Annotations](https://docs.google.com/document/d/17HRBbURSjLThnotRsez2YY0goCF9REb4LEELrpRs7k8/edit?usp=sharing).
